types/src/jwks/mod.rs

```diff
@@ -339,14 +339,12 @@ impl KeyLevelUpdate {
         }
     }
 
-    pub fn as_issuer_level_repr(&self) -> anyhow::Result<ProviderJWKs> {
-        let kid_str = String::from_utf8(self.kid.clone())
-            .context("KeyLevelUpdate::as_issuer_level_repr failed on kid")?;
+    pub fn try_as_issuer_level_repr(&self) -> anyhow::Result<ProviderJWKs> {
         let jwk_repr = self.to_upsert.clone().unwrap_or_else(|| {
-            JWK::RSA(RSA_JWK::new_256_aqab(
-                kid_str.as_str(),
-                DELETE_COMMAND_INDICATOR,
-            ))
+            JWK::Unsupported(UnsupportedJWK {
```

Comment:
how does this work during rollout? when some nodes have old behavior and others have new behavior? also before and after the framework upgrade? it feels to me we need some sort of 2 steps release for this?

Reply:
we have a feature flag that gates both the binary and the framework, so none of the changes in PR 1-5 have taken effect. This PR should only affects the gated logic.

I don't see why 2 steps are needed, could you elaborate?

Reply:
oh i see, as long as there's some feature gating it's fine